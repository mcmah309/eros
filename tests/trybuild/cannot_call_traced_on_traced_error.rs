use eros::{traced, IntoConcreteTracedError};



fn main() {
    let error: eros::TracedError = traced!("Error");
    let _error = error.traced();
}
