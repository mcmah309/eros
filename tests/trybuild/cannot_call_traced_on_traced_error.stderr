error[E0597]: `error` does not live long enough
 --> tests/trybuild/cannot_call_traced_on_traced_error.rs:7:18
  |
6 |     let error: eros::TracedError = traced!("Error");
  |         ----- binding `error` declared here
7 |     let _error = error.traced();
  |                  ^^^^^---------
  |                  |
  |                  borrowed value does not live long enough
  |                  argument requires that `error` is borrowed for `'static`
8 | }
  | - `error` dropped here while still borrowed
